### PR TITLE
Add custom order editor functionality to legend items in Bar chart

### PR DESCRIPTION
**Pull request by tangrammer**
- [ ] **Update release notes if necessary**
- [ ] Add i18n

Relates #625 

- Uses `orderedLegend`  feature-flag
- Add https://github.com/clauderic/react-sortable-hoc and https://github.com/sindresorhus/array-move client libs 
- lumen backend changes only related to specs, just to define the data API we expect in visualisation.spec.legend

```js
{visualisation:
 {spec:
  {legend:
   {order:
    {mode:"custom/auto",
     list: ["v1", "v2", "v3"]}}}}}

```
 
# TODO
- [x] client needs to load legend values 
- [x] client needs to send order selection
- [ ] improve css styles

# Current state

Screenshot 2020-08-10 at 17 12 37

### CODE REVIEW

File: client/src/components/charts/LegendsSortable.jsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  return legend;
+};
+
+export const LegendsSortable = ({
+  onChangeSpec,
+  colors,
+  specLegend,
+  visualisation,
+}) => {
+  const sortable = (get(specLegend, 'order.mode') || 'auto') !== 'auto';
```
Comment (muloem) [inline client/src/components/charts/LegendsSortable.jsx:64]: Why not the positive case here? i.e. 

```suggestion
  const sortable = (get(specLegend, 'order.mode') || 'auto') === 'custom';
```

File: client/src/components/charts/LegendsSortable.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    onChangeSpec({ legend });
+  };
+
+  const getColor = (key, index) => (colors && colors[key]) || palette[index];
+
+  // eslint-disable-next-line new-cap
+  const SortableList = SortableContainer(({ legendItems }) =>
+  (
+    <div style={{ marginBottom: '5px' }}>
+      {legendItems.map((value, index) =>
```
Comment (muloem) [inline client/src/components/charts/LegendsSortable.jsx:80]: The way you have structured this gives the impression that within the same list its possible to find sortable and non-sortable items. Is that the case?

Comment (tangrammer) [reply client/src/components/charts/LegendsSortable.jsx:80]: good catch!

Comment (tangrammer) [reply client/src/components/charts/LegendsSortable.jsx]: this should solve it https://github.com/akvo/akvo-lumen/pull/2803/commits/7f0d069d2fbce9d403154988f61d3e6c657b5556

**Pull request merged**